src/main/java/org/apache/commons/collections4/bloomfilters/ProtoBloomFilter.java

```diff
@@ -0,0 +1,558 @@
+
+    /**
+     * An empty ProtoBloomFilter. Used to create empty BloomFilters.
+     */
+    public static final ProtoBloomFilter EMPTY = new ProtoBloomFilter(Collections.emptyList());
+
+    /**
+     * Get a builder .
+     *
+     * @return a new builder.
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /* package private for testing */
+    /**
+     * Constructor
+     *
+     * @param hashes the two longs that were created by the murmur hash function.
+     */
+    ProtoBloomFilter(Collection<Hash> hashes) {
+        this.hashes = new ArrayList<Hash>();
+        this.hashes.addAll(hashes);
+        // sort so compareTo and equals work properly
+        Collections.sort(this.hashes);
+    }
+
+    private Object writeReplace() {
+        return new ProtoSerProxy(this);
+    }
+
+    /**
+     * Get the count of hashed items included in this proto bloom filter.
+     *
+     * @return The number of unique items in this proto filter.
+     */
+    public int getItemCount() {
+        return hashes.size();
+    }
+
+    /**
+     * Get the stream of hashes included in this proto bloom filter.
+     *
+     * @return the stream of hashes.
+     */
+    public Stream<Hash> getHashes() {
+        return hashes.stream();
+    }
+
+    /**
+     * Get the count of unique hashed items included in this proto bloom filter.
+     *
+     * @return the number of unique items hashed into the proto bloom filter.
+     */
+    public int getUniqueItemCount() {
+        return (int) getUniqueHashes().count();
+    }
+
+    /**
+     * Get the stream of uniques hashes included in this proto bloom filter.
+     *
+     * @return the stream of unique hashes.
+     */
+    public Stream<Hash> getUniqueHashes() {
+        return hashes.stream().distinct();
+    }
+
+    @Override
+    public int hashCode() {
+        if (hashCode == null) {
+            hashCode = Objects.hash(hashes);
+        }
+        return hashCode.intValue();
+    }
+
+    @Override
+    public int compareTo(ProtoBloomFilter other) {
+        Iterator<Hash> otherIter = other.getHashes().iterator();
+        Iterator<Hash> iter = hashes.iterator();
+        int result;
+        while (iter.hasNext() && otherIter.hasNext()) {
+            result = iter.next().compareTo(otherIter.next());
+            if (result != 0) {
+                return result;
+            }
+        }
+        return otherIter.hasNext() ? -1 : 0;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o instanceof ProtoBloomFilter) {
+            return compareTo((ProtoBloomFilter) o) == 0;
+        }
+        return false;
+    }
+
+    @Override
+    public String toString() {
+        return String.format("ProtoBloomFilter[ %s, %s]", hashes.size(), hashCode());
+    }
+
+    /**
+     * A Serialization proxy for a ProtoBloomFilter.
+     *
+     */
+    private static class ProtoSerProxy implements Serializable {
```

Comment:
No need to abbreviate the class name here, I would spell out Serializable in full.